src/olympia/scanners/admin.py

```diff
@@ -440,15 +444,104 @@ class ScannerRuleAdmin(AbstractScannerRuleAdminMixin, admin.ModelAdmin):
 
 @admin.register(ScannerQueryRule)
 class ScannerQueryRuleAdmin(AbstractScannerRuleAdminMixin, admin.ModelAdmin):
-    list_display = ('name', 'scanner', 'state')
+    list_display = (
+        'name', 'scanner', 'state_with_actions', 'matched_results_link',
+    )
     list_filter = ('state',)
     fields = (
         'scanner',
-        'state',
+        'state_with_actions',
         'name',
         'created',
         'modified',
         'matched_results_link',
-        'definition'
+        'definition',
+    )
+    readonly_fields = (
+        'created', 'modified', 'matched_results_link', 'state_with_actions',
     )
-    readonly_fields = ('created', 'modified', 'matched_results_link', 'state')
+
+    def handle_run(self, request, pk, *args, **kwargs):
+        is_admin = acl.action_allowed(
+            request, amo.permissions.ADMIN_SCANNERS_QUERY)
+        if not is_admin or request.method != 'POST':
+            raise Http404
+
+        rule = self.get_object(request, pk)
+        if rule.state == NEW:
+            # Update state right away for the UI to be up to date after we
+            # redirect.
+            rule.update(state=RUNNING)
+            run_yara_query_rule.delay(rule.pk)
+
+            messages.add_message(
+                request,
+                messages.INFO,
+                'Scanner Query Rule {} has been successfully queued for '
+                'execution.'.format(rule.pk),
+            )
+        else:
+            messages.add_message(
+                request,
+                messages.ERROR,
+                'Scanner Query Rule {} could not be queued for execution '
+                'because it was in "{}"" state.'.format(
+                    rule.pk, rule.get_state_display()),
+            )
+
+        return redirect('admin:scanners_scannerqueryrule_changelist')
+
+    def handle_abort(self, request, pk, *args, **kwargs):
+        is_admin = acl.action_allowed(
+            request, amo.permissions.ADMIN_SCANNERS_QUERY)
+        if not is_admin or request.method != 'POST':
+            raise Http404
+
+        rule = self.get_object(request, pk)
+        if rule.state == RUNNING:
+            # FIXME: revoke existing tasks (would need to extract the
+            # GroupResult when executing the chord, store its id in the rule,
+            # then restore the GroupResult here to call revoke() on it)
```

Comment:
Left this as a `FIXME` because I intend to come back to it, I just couldn't get it to work right away. So with this patch the tasks are still there, they just do nothing because they check the state of the rule before running it on the versions chunk each task is responsible for.